src/sympc/grads/grad_functions.py

```diff
@@ -351,6 +351,44 @@ def backward(ctx: Dict[str, Any], grad: MPCTensor) -> MPCTensor:
         return x ** (y - 1) * y * grad
 
 
+class GradDiv(GradFunc):
+    """The Division gradient function."""
+
+    @staticmethod
+    def forward(ctx: Dict[str, Any], x: MPCTensor, y: Any) -> MPCTensor:
+        """Perform the feedforward and compute the result for the division operation.
+
+        Args:
+            ctx (Dict[str, Any]): Context used to save information needed in the backward pass
+            x (MPCTensor): 1st operand for the division operation
+            y (Any): 2nd operand for the divison operation
+
+        Returns:
+            x / y (MPCTensor): The result of the division
+        """
+        ctx["x"] = x
+        ctx["y"] = y
+        return x / y
+
+    @staticmethod
+    def backward(ctx: Dict[str, Any], grad: MPCTensor) -> MPCTensor:
+        """Perform the backward pass for the division operation.
+
+        Args:
+            ctx (Dict[str, Any]): Context used to retrieve the information for the backward pass
+            grad (MPCTensor): The gradient that came from the child nodes
+
+        Returns:
+            (x_grad, y_grad) (Tuple[MPCTensor]): The gradients passed to the X and Y nodes.
+        """
+        x, y = ctx["x"], ctx["y"]
+
+        grad_x = grad * (x / (y * x))
```

Comment:
Q: Why does it look like this?
```
dx x/y = 1/y
```
So for this, you will have `grad / y `.

Reply:
I dont understand this diff :/

Reply:
Good question. I have no idea :P but I am figuring it out. 

I know grad(x/y) wrt to x is (1/x). 
When I first made it (1/y) and compared results with PyTorch for the same, they were very different. But, with some random experimentation I found that (x/(x*y) matches Pytorch's grad for division.

Reply:
Nevermind this. I will correct it. I made a mistake.